### PR TITLE
drop extension / update crd

### PR DESCRIPTION
**Pull request by ChuckHend**
Operator creates/drops postgres extensions.

Updated the CRD for extensions to be a list of objects, instead of a list of strings. Each object has as attribute: `enabled`. Operator will use that attribute to switch its behavior.

### CODE REVIEW

File: coredb-operator/tests/integration_tests.rs [hunk 1]
```diff
@@ -255,6 +255,42 @@ mod test {
...
+        // Apply crd with extension disabled
+        let params = PatchParams::apply("coredb-integration-test");
+        let patch = Patch::Apply(&coredb_json);
+        let coredb_resource = coredbs.patch(name, &params, &patch).await.unwrap();
+
+        // give it time to drop
+        thread::sleep(Duration::from_millis(5000));
+
+        // Assert extension no longer created
+        // Assert extension 'postgis' was created
```
Comment (ChuckHend) [inline coredb-operator/tests/integration_tests.rs:281]:
```suggestion
```

File: coredb-operator/src/controller.rs [hunk 2]
```diff
@@ -86,6 +86,25 @@ pub struct Context {
...
+impl Default for Extension {
+    fn default() -> Self {
+        Extension {
+            name: "pg_stat_statements".to_owned(),
+            version: "1.9".to_owned(),
+            enabled: true,
+            schema: "postgres".to_owned(),
+        }
+    }
+}
```
Comment (ianstanton) [inline coredb-operator/src/controller.rs:106]:
```suggestion
```
I think the default should be an empty vec.

File: coredb-operator/src/defaults.rs [hunk 3]
```diff
@@ -44,6 +46,6 @@ pub fn default_postgres_exporter_image() -> String {
     "quay.io/prometheuscommunity/postgres-exporter:v0.11.1".to_owned()
 }
 
-pub fn default_extensions() -> Vec<String> {
-    vec![]
+pub fn default_extensions() -> Vec<Extension> {
+    vec![Extension::default()]
```
Comment (ianstanton) [inline coredb-operator/src/defaults.rs:50]:
```suggestion
    vec![]
```
Same as above, I think we should default to empty `Vec<Extension>`.

File: coredb-operator/src/extensions.rs [hunk 4]
```diff
@@ -1,32 +1,50 @@
...
-                )
-                .await
-                .unwrap();
-            debug!("Result: {}", result.stdout.clone().unwrap());
+            if ext.enabled {
+                info!("Creating extension: {}", ext_name);
+                // this will no-op if we've already created the extension
+                let result = cdb
+                    .psql(
+                        format!("CREATE EXTENSION IF NOT EXISTS {ext_name};"),
```
Comment (ianstanton) [inline coredb-operator/src/extensions.rs:28]: :ok_hand:

File: coredb-operator/src/extensions.rs [hunk 5]
```diff
@@ -1,32 +1,50 @@
...
+                        "postgres".to_owned(),
+                        client.clone(),
+                    )
+                    .await
+                    .unwrap();
+                debug!("Result: {}", result.stdout.clone().unwrap());
+            } else {
+                info!("Dropping extension: {}", ext_name);
+                // this will no-op if we've already created the extension
```
Comment (ianstanton) [inline coredb-operator/src/extensions.rs:37]:
```suggestion
```

**Review by ianstanton — CHANGES_REQUESTED** [review]

File: coredb-operator/src/extensions.rs (hunk 5 above)
Comment (ChuckHend) [reply coredb-operator/src/extensions.rs:37]: sure, this was left over from previous implementation!

File: coredb-operator/src/controller.rs (hunk 2 above)
Comment (ianstanton) [reply coredb-operator/src/controller.rs:106]: What's an example case for using this? I could see us using it as a default CRD value if we wanted this extension enabled whenever the user provides nothing in `spec.extensions` (like you had previously configured), but we want that to be empty in that case.

Comment (ChuckHend) [reply coredb-operator/src/controller.rs:106]: I use `::Default()` a lot in tests or wherever I want to construct a default set of values for struct.

Comment (ianstanton) [reply coredb-operator/src/controller.rs:106]: :+1: seems reasonable!

**Review by ianstanton — APPROVED** [review]

File: coredb-operator/src/controller.rs (hunk 2 above)
Comment (ChuckHend) [reply coredb-operator/src/controller.rs:106]: `default_extensions` is an empty vec regardless of implementing Default on the Extension

**Review by sjmiller609 — APPROVED** [review]

**Pull request merged**